dspace-server-webapp/src/test/java/org/dspace/app/rest/BrowsesResourceControllerIT.java

```diff
@@ -215,6 +223,108 @@ public void findBrowseBySubjectEntries() throws Exception {
+        Collection col2 = CollectionBuilder.createCollection(context, child1).withName("Collection 2").build();
+
+        //2. Three public items that are readable by Anonymous with different subjects
+        Item publicItem1 = ItemBuilder.createItem(context, col1)
+                                      .withTitle("Public item 1")
+                                      .withIssueDate("2017-10-17")
+                                      .withAuthor("Smith, Donald").withAuthor("Doe, John")
+                                      .withSubject("History of religion", "VR110102", 600)
+                                      .build();
+
+        Item publicItem2 = ItemBuilder.createItem(context, col2)
+                                      .withTitle("Public item 2")
+                                      .withIssueDate("2016-02-13")
+                                      .withAuthor("Smith, Maria").withAuthor("Doe, Jane")
+                                      .withSubject("Church studies", "VR110103", 600)
+                                      .withSubject("History of religion", "VR110102", 600)
+                                      .build();
+
+        Item publicItem3 = ItemBuilder.createItem(context, col2)
+                                      .withTitle("Public item 2")
+                                      .withIssueDate("2016-02-13")
+                                      .withAuthor("Smith, Maria").withAuthor("Doe, Jane")
+                                      .withSubject("Missionary studies", "VR110104", 600)
+                                      .withSubject("Church studies", "VR110103", 600)
+                                      .withSubject("History of religion", "VR110102", 600)
+                                      .build();
+
+        context.restoreAuthSystemState();
+
+        //** WHEN **
+        //An anonymous user browses this endpoint to find which subjects are currently in the repository
+        getClient().perform(get("/api/discover/browses/subject/entries")
+                                .param("projection", "full"))
+
+                   //** THEN **
+                   //The status has to be 200
+                   .andExpect(status().isOk())
+
+                   //We expect the content type to be "application/hal+json;charset=UTF-8"
+                   .andExpect(content().contentType(contentType))
+                   .andExpect(jsonPath("$.page.size", is(20)))
+                   //Check that there are indeed 3 different subjects
+                   .andExpect(jsonPath("$.page.totalElements", is(3)))
+                   //Check the embedded resources and that they're sorted alphabetically
+                   //Check that the subject matches as expected
+                   //Verify that they're sorted alphabetically
+                   .andExpect(
+                       jsonPath("$._embedded.entries",
+                            containsInAnyOrder(
```

Comment:
In the comment above you mention that the embedded resources should be "sorted alphabetically"...but here you are using `containsInAnyOrder` which doesn't check sort order obviously.  If they are expected in a specific order, we should verify the exact order as well.